### DISCUSSION TITLE
Add "Deprecated" field meta tag

### DISCUSSION THREAD

**Discussion by stafyniaksacha** [directus/directus#23768] [Feature Requests]
### Summary

Add a checkbox in directus field edition form to mark it as deprecated.

When deprecated, the field is still viewable/editable in content editor, but user as feedback that the field should not be used (field name is struck, or an icon is present) 

### Motivation

Deprecating the field can be useful when performing long-running migrations, or when we want to support legacy features without hiding them.

### Detailed Design

- add `deprecated` column in `directus_field`
- update `FieldMeta` schema
- update app settings module to be able to mark field as deprecated
- update app content module to display deprecated field

### Could Have:
- A deprecated reason would be a good addition

**Comment by FreeSynergy** [comment]
Good Idea +1

**Comment by remihuigen** [comment]
In addition to that, it would also be a great addition for the generated openapi specs if we get set the [`deprecated` property](https://arc.net/l/quote/zvqrbrgd). Currently, we have no clear way of communicating to third party api consumers that a field is considered 'deprecated'
